### PR TITLE
Feat : Home,commit history,dc modal

### LINKED ISSUE #4: Commit-History,Home,DC Modal
**[Figma Link](https://www.figma.com/file/yO3edbsblDYwj42L8yYLrI/CodeCharacter-2023?node-id=80%3A120&t=GTLykSBvdJnUQFqI-0)**
## Add Commit History Page
- Make a ```vertical timeline``` displaying all commits.
- Create a slide toggle button for ```map``` and ```code``` commits
- For displaying code in the ```Editor``` view use [react-code-blocks](https://www.npmjs.com/package/react-code-blocks)
- For displaying map in the ```Map``` view, display it as an image
- Integrate with backend
## Add Home page
- Implement a good particle effect for background
## DC-Modal
- Create a simple ```modal``` for displaying Daily-Challenges
- Integrate with backend

### PR DESCRIPTION
**Pull request by shubham-1806**
To Do :
- [x] Add Particle Effect

Fixes #4

### CODE REVIEW

File: src/components/Home/Glitchtext/Glitchtext.module.css [hunk 1]
```diff
@@ -10,10 +10,10 @@
 .glitchtext {
   text-align: center;
   color: #fff;
-  font-size: 90px;
+  font-size: 120px;
   letter-spacing: 4px;
   font-weight: 800;
-  font-family: 'Montserrat';
+  font-family: 'Bai Jamjuree';
```
Comment (Ram-20062003) [inline src/components/Home/Glitchtext/Glitchtext.module.css:16]: You have download and import this from google-fonts.

File: src/pages/Home/Home.tsx [hunk 2]
```diff
@@ -1,15 +1,16 @@
...
-import Rows from '../../components/Home/Row/Rows';
+import AboutGame from '../../components/Home/AboutGame/AboutGame';
 import styles from './Home.module.css';
 
 export default function Home(): JSX.Element {
   return (
     <main className={styles.mainContainer}>
       <div className={styles.jumbotron}>
         <Glitchtext value="Code Character" />
+        <img className={styles.arrow} src="/assets/DownArrow.svg"></img>
```
Comment (Ram-20062003) [inline src/pages/Home/Home.tsx]: When the ```arrow``` is clicked it has to move second page segment.

File: src/components/CommitHistory/CodeMapViewbox/CodeView.tsx [hunk 3]
```diff
@@ -1,15 +1,16 @@
...
-import cpp from 'react-syntax-highlighter/dist/esm/languages/prism/cpp';
-import python from 'react-syntax-highlighter/dist/esm/languages/prism/python';
-import java from 'react-syntax-highlighter/dist/esm/languages/prism/java';
-import { atomDark } from 'react-syntax-highlighter/dist/esm/styles/prism';
+// import { PrismLight as SyntaxHighlighter } from 'react-syntax-highlighter';
+// import c from 'react-syntax-highlighter/dist/esm/languages/prism/c';
+// import cpp from 'react-syntax-highlighter/dist/esm/languages/prism/cpp';
+// import python from 'react-syntax-highlighter/dist/esm/languages/prism/python';
+// import java from 'react-syntax-highlighter/dist/esm/languages/prism/java';
+// import { atomDark } from 'react-syntax-highlighter/dist/esm/styles/prism';
```
Comment (Ram-20062003) [inline src/components/CommitHistory/CodeMapViewbox/CodeView.tsx]: Remove these

File: src/components/CommitHistory/CodeMapViewbox/CodeView.tsx [hunk 4]
```diff
@@ -1,15 +1,16 @@
...
+import { CodeBlock, atomOneDark } from 'react-code-blocks';
```
Comment (Ram-20062003) [inline src/components/CommitHistory/CodeMapViewbox/CodeView.tsx]: `atomOneDark` to `iRBlack` or something darker matching to figma

File: src/components/CommitHistory/CodeMapViewbox/CodeView.tsx [hunk 5]
```diff
@@ -1,15 +1,16 @@
...
 
-SyntaxHighlighter.registerLanguage('c', c);
-SyntaxHighlighter.registerLanguage('cpp', cpp);
-SyntaxHighlighter.registerLanguage('python', python);
-SyntaxHighlighter.registerLanguage('java', java);
+// SyntaxHighlighter.registerLanguage('c', c);
+// SyntaxHighlighter.registerLanguage('cpp', cpp);
+// SyntaxHighlighter.registerLanguage('python', python);
+// SyntaxHighlighter.registerLanguage('java', java);
```
Comment (Ram-20062003) [inline src/components/CommitHistory/CodeMapViewbox/CodeView.tsx]: Remove

File: src/components/CommitHistory/CodeMapViewbox/CodeView.tsx [hunk 6]
```diff
@@ -18,12 +19,18 @@ type PropsType = {
...
 export default function CodeView(props: PropsType): JSX.Element {
   return (
-    <SyntaxHighlighter
+    // <SyntaxHighlighter
+    //   language={props.codeLang}
+    //   style={atomDark}
+    //   showLineNumbers={props.code != '' ? true : false}
+    // >
+    //   {props.code}
+    // </SyntaxHighlighter>
```
Comment (Ram-20062003) [inline src/components/CommitHistory/CodeMapViewbox/CodeView.tsx]: Remove comments

File: src/components/CommitHistory/HistoryMain/History.module.css [hunk 7]
```diff
@@ -22,33 +25,45 @@
...
-.largeButton {
+.whiteButton {
   align-content: right;
-  color: #ffffff;
-  font-weight: 600;
-  font-family: Montserrat;
+  color: #ffffff !important;
+  font-weight: 400;
+  font-family: Bai Jamjuree;
   font-style: normal;
```
Comment (Ram-20062003) [inline src/components/CommitHistory/HistoryMain/History.module.css:34]: Font is not properly imported here

File: src/components/CommitHistory/HistoryMain/History.module.css [hunk 8]
```diff
@@ -22,33 +25,45 @@
...
-.smallButton {
+.darkButton {
   align-content: right;
-  color: #ffffff;
-  background: #1a3051;
-  font-family: Montserrat;
-  font-weight: 600;
+  color: rgba(234, 234, 234, 0.6);
+  background-color: inherit !important;
+  font-family: Bai Jamjuree;
```
Comment (Ram-20062003) [inline src/components/CommitHistory/HistoryMain/History.module.css:49]: here too

File: src/components/CommitHistory/HistoryMain/History.module.css [hunk 9]
```diff
@@ -90,16 +105,16 @@
...
-  font-size: 22px;
+  font-size: 20px;
   font-style: normal;
-  background-color: #48c6d8;
-  border: 0px;
-  border-radius: 2px;
+  background-color: inherit;
+  border: 1px solid white;
+  border-radius: 5px;
+  font-family: Bai Jamjuree;
```
Comment (Ram-20062003) [inline src/components/CommitHistory/HistoryMain/History.module.css:113]: Font is not properly imported here

File: src/components/NavBar/NavBar.module.css [hunk 10]
```diff
@@ -81,4 +101,24 @@
...
+
+.dcIcon {
+  height: 2rem;
+  width: auto;
+  margin: auto;
+  position: relative;
+  margin-right: 2rem;
+}
+
+.modal{
```
Comment (Ram-20062003) [inline src/components/NavBar/NavBar.module.css]: `modal` -> `dcmodal`

**Review by Ram-20062003 — CHANGES_REQUESTED** [review]
## Changes Requested
_Please follow the figma provided_
### Home-Page
- [x]  Import fonts from `google-fonts` and then use it . `Bai Jamjuree` will not be present by default
- [x] Change the `Footer` in the front-page. Change the design according to our `new colour` palette
- [x] Giver darker bg for **Made with :heart: by delta**. Match it with our new design.
- [x] Look into responsiveness, 
      &emsp;1. `Text overflow` in About page for mobiles.
      &emsp;2. Reduce `font-size` accordingly
      &emsp;3. Have little margin 
- [x] The `arrow` in the home page should move to second segment when clicked
###  Commit History Page
- [x]  Use proper fonts here
- [x] Change the react-code-blocks theme are from `aotmonedark` to `iRBlack`
- [x] The commit msg area box should be of `uniform size`. Should **not** change for `longer commit messages`.
- [x] Follow `proper dimensions and colour` as mentioned in figma
- [x] Vertical timeline component boxes are supposed to be `translucent`
- [x] Upon clicking `Load Commit`, the committed code should be loaded in main editor
- [x] Make sure the page is responsive upto `13.1 inches` laptop (1248px).
### Daily Challenge
- [x] Change the color pallete of the dropdown. Match it with the theme
### General Changes
- [x] Remove commented and redundant codes
- [x] Give proper names to the variables
- [x] Figure out a proper particle background.
- [x] Remove unwanted files too

Please get these changes done asap. If you have any doubts let me know

File: src/components/NavBar/NavBar.tsx [hunk 11]
```diff
@@ -85,18 +85,16 @@ const NavBar: React.FunctionComponent = () => {
...
   const handleCloseCompleted = () => setShowCompleted(false);
-  // const handleShowCompleted = () => setShowCompleted(true);
 
   const [showAvailable, setShowAvailable] = useState(false);
   const handleCloseAvailable = () => setShowAvailable(false);
-  // const handleShowAvailable = () => setShowAvailable(true);
 
   return (
     <div className={styles.navBar}>
       <Modal
```
Comment (Ram-20062003) [inline src/components/NavBar/NavBar.tsx]: Make separate these modals as components and import it as a component here

File: src/components/NavBar/NavBar.tsx [hunk 12]
```diff
@@ -105,18 +103,18 @@ const NavBar: React.FunctionComponent = () => {
...
+          <Button className={styles.dcmodalBtn} onClick={handleCloseCompleted}>
             Nice!
           </Button>
-          <Button variant="primary" onClick={handleCloseCompleted}>
+          <Button className={styles.dcmodalBtn} onClick={handleCloseCompleted}>
             View challenges
           </Button>
         </Modal.Footer>
       </Modal>
       <Modal
```
Comment (Ram-20062003) [inline src/components/NavBar/NavBar.tsx]: Make a separate component for this Modal too

File: src/components/NavBar/NavBar.tsx [hunk 13]
```diff
@@ -125,10 +123,10 @@ const NavBar: React.FunctionComponent = () => {
           Commander we hope you are ready for today&apos;s daily chellenge
         </Modal.Body>
         <Modal.Footer>
-          <Button variant="secondary" onClick={handleCloseAvailable}>
+          <Button className={styles.dcmodalBtn} onClick={handleCloseAvailable}>
```
Comment (Ram-20062003) [inline src/components/NavBar/NavBar.tsx]: Refer to [react-bootstrap](https://react-bootstrap.github.io/components/buttons/)
```suggestion
<Button variant="outline-danger" onClick={handleCloseAvailable}>
```

File: src/components/NavBar/NavBar.tsx [hunk 14]
```diff
@@ -125,10 +123,10 @@ const NavBar: React.FunctionComponent = () => {
...
             Not Right Now!
           </Button>
-          <Button variant="primary" onClick={handleCloseAvailable}>
+          <Button className={styles.dcmodalBtn} onClick={handleCloseAvailable}>
```
Comment (Ram-20062003) [inline src/components/NavBar/NavBar.tsx]: Refer to [react-bootstrap](https://react-bootstrap.github.io/components/buttons/)
```suggestion
<Button variant="outline-light" onClick={handleCloseAvailable}>
```

File: src/components/NavBar/NavBar.tsx [hunk 15]
```diff
@@ -105,18 +103,18 @@ const NavBar: React.FunctionComponent = () => {
           Woohoo, youv&apos;e completed today&apos;s challenges!
         </Modal.Body>
         <Modal.Footer>
-          <Button variant="secondary" onClick={handleCloseCompleted}>
+          <Button className={styles.dcmodalBtn} onClick={handleCloseCompleted}>
             Nice!
           </Button>
```
Comment (Ram-20062003) [inline src/components/NavBar/NavBar.tsx]: - What does this button does here other than closing the modal? Is `Nice` just like `cancel` button.
- `Nice` -> `Ok`
- Refer to [react-bootstrap](https://react-bootstrap.github.io/components/buttons/) and have an `variant` property for it. Make sure it is an `outline` propety

File: src/components/NavBar/NavBar.tsx [hunk 16]
```diff
@@ -105,18 +103,18 @@ const NavBar: React.FunctionComponent = () => {
...
-          <Button variant="primary" onClick={handleCloseCompleted}>
+          <Button className={styles.dcmodalBtn} onClick={handleCloseCompleted}>
```
Comment (Ram-20062003) [inline src/components/NavBar/NavBar.tsx]:
```suggestion
<Button variant="outline-info" onClick={handleCloseAvailable}>
```

File: src/components/NavBar/NavBar.tsx [hunk 17]
```diff
@@ -105,18 +103,18 @@ const NavBar: React.FunctionComponent = () => {
...
             View challenges
```
Comment (Ram-20062003) [inline src/components/NavBar/NavBar.tsx]: `View Challenges` -> `View LeaderBoard` . We can redirect him to `daily-challenge leaderboard`.
_(DC-LeaderBoard is not yet ready, you can add the calling latter)_

File: src/components/NavBar/NavBar.tsx [hunk 18]
```diff
@@ -125,10 +123,10 @@ const NavBar: React.FunctionComponent = () => {
           Commander we hope you are ready for today&apos;s daily chellenge
```
Comment (Ram-20062003) [inline src/components/NavBar/NavBar.tsx]:
```suggestion
Commander we hope you are ready for today&apos;s Daily Challenge
```

File: src/components/CommitHistory/HistoryMain/History.tsx [hunk 19]
```diff
@@ -78,11 +79,20 @@ export default function History(): JSX.Element {
...
+          currentUserCode: currentCode,
+        }),
+      );
+      toast.success('Loaded');
     } else if (SelectedButton == 'Map' && currentMap.length != 0) {
       dispatch(changeHistoryEditorMap(currentMap));
     }
   };
 
+  console.log(currentCode);
```
Comment (Ram-20062003) [inline src/components/CommitHistory/HistoryMain/History.tsx]: Remove logs

**Review by Ram-20062003 — CHANGES_REQUESTED** [review]
## UI-Changes

### Commit-History Page
image
- [x] Limit the height accordingly to the map view box height

### Home-Page
- [x] Replace the image with *previous year map image*
image
- [x] We can have `video-loop` background here too.

### General Changes
- [x] Play the `video-loop` slower (0.2x) and once the video is done play it in `reverse` direction
- [x] Remove these un-necessary files. [model.json](https://github.com/delta/codecharacter-web-2023/blob/home%2CcommitHistory%2CDCModal/models.json) and [binary](https://github.com/delta/codecharacter-web-2023/blob/home%2CcommitHistory%2CDCModal/1A3F919CD5B7725108B03C349630A1C82B03_4A84987F2428442A9EC84563BB2BC063)

_(Make sure to do these and the below changes @shubham-1806)_

File: src/components/CommitHistory/HistoryMain/History.tsx [hunk 20]
```diff
@@ -78,14 +80,28 @@ export default function History(): JSX.Element {
...
+        case 'python':
+          localStorage.setItem('languageChose', 'Python');
+          break;
+        case 'java':
+          localStorage.setItem('languageChose', 'Java');
+          break;
+        default:
+          dispatch(changeLanguage('c_cpp'));
+      }
+      toast.success(` Loaded commit -> ${currentCommitMessage}`);
```
Comment (Ram-20062003) [inline src/components/CommitHistory/HistoryMain/History.tsx]:
``` suggestion
      toast.success(` Loaded commit - ${currentCommitMessage}`);
```

**Review by Ram-20062003 — APPROVED** [review]

**Review by vigneshd332 — APPROVED** [review]

**Pull request merged**